tfhe/src/integer/client_key/mod.rs

```diff
@@ -300,6 +280,152 @@ impl ClientKey {
         recomposer.value()
     }
 
+    pub fn encrypt_signed_radix<T>(&self, message: T, num_blocks: usize) -> SignedRadixCiphertext
+    where
+        T: DecomposableInto<u64> + SignedNumeric,
+    {
+        encrypt_words_radix_impl(
+            &self.key,
+            message,
+            num_blocks,
+            crate::shortint::ClientKey::encrypt,
+        )
+    }
+
+    pub fn encrypt_signed_radix_without_padding<T>(
+        &self,
+        message: T,
+        num_blocks: usize,
+    ) -> SignedRadixCiphertext
+    where
+        T: DecomposableInto<u64> + SignedNumeric,
+    {
+        encrypt_words_radix_impl(
+            &self.key,
+            message,
+            num_blocks,
+            crate::shortint::ClientKey::encrypt_without_padding,
+        )
+    }
+
+    pub fn encrypt_signed_radix_compressed<T: DecomposableInto<u64> + SignedNumeric>(
+        &self,
+        message: T,
+        num_blocks: usize,
+    ) -> CompressedSignedRadixCiphertext {
+        encrypt_words_radix_impl(
+            &self.key,
+            message,
+            num_blocks,
+            crate::shortint::ClientKey::encrypt_compressed,
+        )
+    }
+
+    pub fn encrypt_signed_radix_without_padding_compressed<
+        T: DecomposableInto<u64> + SignedNumeric,
+    >(
+        &self,
+        message: T,
+        num_blocks: usize,
+    ) -> CompressedSignedRadixCiphertext {
+        encrypt_words_radix_impl(
+            &self.key,
+            message,
+            num_blocks,
+            crate::shortint::ClientKey::encrypt_without_padding_compressed,
+        )
+    }
+
+    pub fn decrypt_signed_radix<T>(&self, ctxt: &SignedRadixCiphertext) -> T
+    where
+        T: RecomposableFrom<u64>
+            + std::ops::Neg<Output = T>
+            + std::ops::Shr<u32, Output = T>
+            + std::ops::BitOrAssign<T>
+            + std::ops::BitOr<T, Output = T>
+            + std::ops::Mul<T, Output = T>
+            + CastFrom<T>
+            + SignedNumeric,
```

Comment:
non trivial bound, can we do something about it ?

Reply:
a trait name `RecomposableSignedInteger`  ? `RecomposableFrom<u64> + Signedinteger` could work, but means would need to wait until our I256 trait can implement it to be able to encrypt/decrypt I256 values